assets/src/edit-story/components/checklist/checks/videoOptimization.js

```diff
@@ -0,0 +1,151 @@
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { __ } from '@web-stories-wp/i18n';
+import styled from 'styled-components';
+
+/**
+ * Internal dependencies
+ */
+import { useCallback, useMemo, useState } from 'react';
+import { useLocalMedia, useStory } from '../../../app';
+import { VIDEO_SIZE_THRESHOLD } from '../../../app/media/utils/useFFmpeg';
+import { PRIORITY_COPY } from '../constants';
+import { LayerThumbnail, Thumbnail, THUMBNAIL_TYPES } from '../../thumbnail';
+import {
+  CARD_TYPE,
+  ChecklistCard,
+  DefaultFooterText,
+} from '../../checklistCard';
+import { filterStoryElements } from '../utils/filterStoryElements';
+import { Button, BUTTON_SIZES, BUTTON_TYPES } from '../../../../design-system';
+import { useHighlights } from '../../../app/highlights';
+
+const OptimizeButton = styled(Button)`
+  margin-top: 4px;
+  border: ${({ theme }) => `1px solid ${theme.colors.border.defaultNormal}`};
+`;
+
+export function videoElementOptimized(element = {}) {
+  if (element.resource?.isTranscoding) {
+    return true;
+  }
+
+  if (
+    element.type !== 'video' ||
+    element.resource?.local ||
+    element.resource?.isOptimized
+  ) {
+    return false;
+  }
+
+  const videoArea =
+    (element.resource?.height ?? 0) * (element.resource?.width ?? 0);
+  const isLargeVideo =
+    videoArea >= VIDEO_SIZE_THRESHOLD.WIDTH * VIDEO_SIZE_THRESHOLD.HEIGHT;
+  return isLargeVideo;
+}
+
+export const BulkVideoOptimization = () => {
+  const story = useStory(({ state }) => state);
+  const unoptimizedVideos = filterStoryElements(story, videoElementOptimized);
+
+  const setHighlights = useHighlights(({ setHighlights }) => setHighlights);
+
+  const { optimizeVideo } = useLocalMedia((state) => ({
+    optimizeVideo: state.actions.optimizeVideo,
+  }));
+
+  const [isOptimizing, setIsOptimizing] = useState(false);
+
+  const sequencedVideoOptimization = useCallback(() => {
+    return new Promise((res) => {
+      unoptimizedVideos
+        .filter(({ resource }) => !resource?.isTranscoding)
+        .map(({ resource }) => async () => {
+          await optimizeVideo({ resource });
+        })
+        .reduce((p, fn) => p.then(fn), Promise.resolve())
+        .then(() => {
+          res();
+        });
+    });
+  }, [unoptimizedVideos, optimizeVideo]);
+
+  const handleUpdateVideos = useCallback(async () => {
+    setIsOptimizing(true);
+    await sequencedVideoOptimization();
+    setIsOptimizing(false);
+  }, [sequencedVideoOptimization]);
+
+  const { footer, title } = PRIORITY_COPY.videoNotOptimized;
+
+  const isTranscoding = useMemo(
+    () =>
+      isOptimizing ||
+      unoptimizedVideos.some((video) => video.resource?.isTranscoding),
+    [isOptimizing, unoptimizedVideos]
+  );
+  return (
+    unoptimizedVideos.length > 0 && (
+      <ChecklistCard
+        cta={
+          <OptimizeButton
+            type={BUTTON_TYPES.TERTIARY}
```

Comment:
Figma has this listed as the `SECONDARY` button? 
<img width="592" alt="Screen Shot 2021-06-25 at 9 03 18 AM" src="https://user-images.githubusercontent.com/10720454/123453283-31a97f80-d594-11eb-918e-7c78dd06e3fe.png">

Reply:
yeah you should be able to just swap this to secondary and eliminate the custom border you added to the button.
![Screen Shot 2021-06-25 at 9 40 47 AM](https://user-images.githubusercontent.com/10720454/123458109-81d71080-d599-11eb-8bf1-bd3dea142e91.png)